packages/puppeteer-core/src/util/disposable.ts

```diff
@@ -261,15 +285,83 @@ export class AsyncDisposableStack {
    */
   move(): AsyncDisposableStack {
     if (this.#disposed) {
-      throw new ReferenceError('a disposed stack can not use anything new'); // step 3
+      throw new ReferenceError('A disposed stack can not use anything new');
     }
-    const stack = new AsyncDisposableStack(); // step 4-5
+    const stack = new AsyncDisposableStack();
     stack.#stack = this.#stack;
+    this.#stack = [];
     this.#disposed = true;
     return stack;
   }
 
-  [asyncDisposeSymbol] = this.dispose;
+  /**
+   * Disposes each resource in the stack in last-in-first-out (LIFO) manner.
+   */
+  async [asyncDisposeSymbol](): Promise<void> {
+    if (this.#disposed) {
+      return;
+    }
+    this.#disposed = true;
+    const errors: unknown[] = [];
+    for (const resource of this.#stack.reverse()) {
+      try {
+        await resource[asyncDisposeSymbol]();
+      } catch (e) {
+        errors.push(e);
+      }
+    }
+    if (errors.length === 1) {
+      throw errors[0];
+    } else if (errors.length > 1) {
+      let suppressed = null;
+      for (const error of errors.reverse()) {
+        if (suppressed === null) {
+          suppressed = error;
+        } else {
+          suppressed = new SuppressedError(error, suppressed);
+        }
+      }
+      throw suppressed;
+    }
+  }
 
   readonly [Symbol.toStringTag] = 'AsyncDisposableStack';
 }
+
+/**
+ * Represents an error that occurs when multiple errors are thrown during
+ * the disposal of resources. This class encapsulates the primary error and
+ * any suppressed errors that occurred subsequently.
```

Comment:
Needs to be marked as internal in the comment, so it does not show up in the docs.
```suggestion
 * any suppressed errors that occurred subsequently.
 * @internal
```
Remember to run `npm run docs` after that.

Reply:
Oh yeah missed that, let me fix it

Reply:
Fixed